host/sgx/create.c

```diff
@@ -560,6 +560,9 @@ oe_result_t oe_sgx_build_enclave(
     enclave->size = enclave_size;
     enclave->text = enclave_addr + oeimage.text_rva;
 
+    // Set the XFRM field in properties structure
+    props.config.xfrm = context->attributes.xfrm;
```

Comment:
It took me a bit to figure out why this declaration is embedded here. You've introduced additional complexity by populating different parts of the `context->attribute` struct at different points in the code and that has downstream effects. 

It doesn't look like there's any particular reason to delay the determination of the XFRM value until `_new_secs`. I suggest:

- [ ] Move the XFRM initialization code from `_new_secs` into `oe_sgx_initialize_load_context`.
    - Consider abstracting the XFRM detection code into its own method that can be hooked for testing against legacy and AVX configurations.
- [ ] Pass the `context` into `_new_secs` instead of just the `debug` flag.
- [ ] Group the initialization of the `props.config.xfrm` with the checking against `props.config.attributes`

Reply:
ok.

Reply:
Done.